gdnative-core/src/geom/basis.rs

```diff
@@ -29,14 +116,316 @@ impl Basis {
         }
     }
 
-    pub fn from_diagonal(p_diag: Vector3) -> Basis {
-        Basis {
-            elements: [
-                Vector3::new(p_diag.x, 0.0, 0.0),
-                Vector3::new(0.0, p_diag.y, 0.0),
-                Vector3::new(0.0, 0.0, p_diag.z),
-            ],
+    /// Inverts the matrix.
```

Comment:
This method may need a `Panics` section, seeing that it can panic if the determinant is zero:

```rust
/// ...
///
/// # Panics
///
/// If the determinant of `self` is zero.
```

This also applies to `inverted`.